repair/row_level.cc

```diff
@@ -1021,13 +1021,14 @@ class repair_meta {
 
     future<uint64_t> do_estimate_partitions_on_local_shard() {
         auto& cf = _db.local().find_column_family(_schema->id());
-        return do_with(cf.get_sstables(), uint64_t(0), [this] (lw_shared_ptr<const sstable_list>& sstables, uint64_t& partition_count) {
-            return do_for_each(*sstables, [this, &partition_count] (const sstables::shared_sstable& sst) mutable {
+        lw_shared_ptr<const sstable_list> sstables = cf.get_sstables();
+        uint64_t partition_count = 0;
+        {
+            for (const sstables::shared_sstable& sst : *sstables) {
                 partition_count += sst->estimated_keys_for_range(_range);
-            }).then([&partition_count] {
-                return partition_count;
-            });
-        });
+            }
+        }
+        co_return partition_count;
```

Comment:
Why is it still a coroutine then? In its do_for_each-based version it could implicitly yield on long loops at least, now it's just a non-preemptive loop

Reply:
Will add a yield. It could stall with Leveled Compaction Strategy.